### PR TITLE
Enable custom bot filtering (Close #121)

### LINKED ISSUE #121: Enable custom bot filtering
### Describe the feature
Users should be able to define their own bot filter criteria.

### Who will this benefit?
Anyone who needs a custom bot filtering option for precision.

### Implementation
By abstracting it away through a macro users could overwrite it to however they prefer, which would give the biggest flexibility with limited impact on current behaviour / username

### PR DESCRIPTION
**Pull request by agnessnowplow**
## Description & motivation
In connection to this [issue](https://snplow.atlassian.net/jira/software/c/projects/PE/issues/PE-3888) 
As the title says to enable custom bot filtering.

## Checklist
- [ ] I have verified that these changes work locally
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: docs/markdown/snowplow_web_overview.md [hunk 1]
```diff
@@ -596,7 +596,15 @@ All the incremental models in the Snowplow web package have recommended cluster
...
+The cluster key macros (see above), the `allow_refresh()` and the `filter_bots` macro can be overridden. These are all [dispatched macros](https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch) and can be overridden by creating your own version of the macro and setting a project level dispatch config. More details can be found in [dbt's docs](https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch#overriding-package-macros)
+
+
+``` yaml
+# Your_dbt_project/macros/filter_bots.sql
+{% macro default__filter_bots() %}
+  and ev.useragent not similar to '%(YOUR_CUSTOM_PATTERN|bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt)%'
+{% endmacro %}
+```
```
Comment (rlh1994) [inline docs/markdown/snowplow_web_overview.md:607]: 🚫
This file is going to be stripped down heavily from #120 , could you raise a PR in https://github.com/snowplow/documentation instead to edit the `docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-advanced-operation/index.md` file and add this there instead?

File: models/page_views/scratch/bigquery/snowplow_web_page_views_this_run.sql [hunk 2]
```diff
@@ -116,7 +116,7 @@ from (
 )
 
 {% if var("snowplow__ua_bot_filter", true) %}
-   where not regexp_contains(ev.useragent, '(bot|crawl|slurp|spider|archiv|spinn|sniff|seo|audit|survey|pingdom|worm|capture|(browser|screen)shots|analyz|index|thumb|check|facebook|PingdomBot|PhantomJS|YandexBot|Twitterbot|a_archiver|facebookexternalhit|Bingbot|BingPreview|Googlebot|Baiduspider|360(Spider|User-agent)|semalt)')
+   {{ filter_bots() }}
 {% endif %}
```
Comment (rlh1994) [inline models/page_views/scratch/bigquery/snowplow_web_page_views_this_run.sql:120]: 🔧 
To increase consistency across the warehouses and avoid issues in the future, would it be worth writing this as 
```sql
where 
   1=1
{% if var("snowplow__ua_bot_filter", true) %}
   {{ filter_bots() }}
{% endif %}
```

and then all of the macros start with `and`?

**Review by rlh1994 — CHANGES_REQUESTED** [review]
Looks good overall, just the docs need to be added to the main docs site from now-on and think there's something we can do to make the approach more consistent across dbs :)

File: docs/markdown/snowplow_web_overview.md (hunk 1 above)
Comment (agnessnowplow) [reply docs/markdown/snowplow_web_overview.md:607]: Sure, we would probably need some marketing around this to highlight to users that the current docs are not the most comprehensive anymore, but then I guess it is part of the upcoming release around that.

File: models/page_views/scratch/bigquery/snowplow_web_page_views_this_run.sql (hunk 2 above)
Comment (agnessnowplow) [reply models/page_views/scratch/bigquery/snowplow_web_page_views_this_run.sql:120]: makes sense, will push it through again with the above change as well.

File: docs/markdown/snowplow_web_overview.md (hunk 1 above)
Comment (rlh1994) [reply docs/markdown/snowplow_web_overview.md]: Yep 100%, the new docs also point to the docs.snowplow.io site as the main source going forward as well once they are merged.

**Review by rlh1994 — APPROVED** [review]

**Pull request merged**